### PR TITLE
Changed "Weyland: Built to last" to be mandatory

### LINKED ISSUE #5654: Weyland: Built To Last shouldn't have a prompt
Currently, Weyland Built To Last is implemented as a "may" ability, but the card image text has a mandatory credit gain when advancing something for the first time.

### LINKED ISSUE #5659: Weyland: Built To Last shouldn't trigger when Masvingo is triggered
Masvingo https://netrunnerdb.com/en/card/21099 
currently triggers Weyland Consortium: Built to Last https://netrunnerdb.com/en/card/30059
but shouldn't

### PR DESCRIPTION
**Pull request by merrickbrown**
fixes #5654 fixes #5659

### CODE REVIEW

File: src/clj/game/cards/identities.clj [hunk 1]
```diff
@@ -1644,20 +1644,14 @@
...
-   :abilities [(set-autoresolve :auto-build-to-last "Built to Last")]})
+             :req (req (not (pos? (- (get-counters target :advancement) (:amount (second targets) 0)))))
+             :msg "gain 2 [Credits]"
+             :effect (req (gain-credits state :corp eid 2))}]})
 
 (defcard "Whizzard: Master Gamer"
   {:recurring 3
    :interactions {:pay-credits {:req (req (and (= :runner-trash-corp-cards (:source-type eid))
-                                               (corp? target)))
+                                            (corp? target)))
```
Comment (NoahTheDuke) [inline src/clj/game/cards/identities.clj:1654]: Looks like indentation got messed up here.

File: src/clj/game/cards/identities.clj [hunk 2]
```diff
@@ -1644,20 +1644,14 @@
...
-              :prompt "Gain 2 [Credits]?"
-              :autoresolve (get-autoresolve :auto-build-to-last)
-              :yes-ability
-              {:async true
-               :msg "gain 2 [Credits]"
-               :effect (req (gain-credits state :corp eid 2))}}}]
-   :abilities [(set-autoresolve :auto-build-to-last "Built to Last")]})
+             :req (req (not (pos? (- (get-counters target :advancement) (:amount (second targets) 0)))))
+             :msg "gain 2 [Credits]"
+             :effect (req (gain-credits state :corp eid 2))}]})
```
Comment (NoahTheDuke) [inline src/clj/game/cards/identities.clj:1649]: This ability needs `:async true`

Comment (merrickbrown) [reply src/clj/game/cards/identities.clj:1649]: Easy enough - probably a bit much for here, but the doc says
```clojure 
;  :async -- boolean
;   Mark the ability as "async", meaning the :effect function must call effect-completed itself.
;   Without this being set to true, resolve-ability will call effect-completed once it's done.
;   This part of the engine is really dumb and complicated, so ask someone on slack about it.```
 Looks like the most relevant code is `engine/do-ability`?

**Comment by merrickbrown** [comment]
May try to address #5659 in this too

**Comment by merrickbrown** [comment]
Fixes #5659

File: src/clj/game/core/props.clj [hunk 3]
```diff
@@ -16,8 +16,7 @@
    (let [updated-card (update! state side (update card prop-type #(+ (or % 0) n)))
          args {:counter-type prop-type :amount n :placed placed}]
      (if (= prop-type :advance-counter)
-       (do (when (and (ice? updated-card)
-                      (rezzed? updated-card))
+       (do (when ((every-pred ice? rezzed?) updated-card)
```
Comment (NoahTheDuke) [inline src/clj/game/core/props.clj:19]: Mind reverting this? `every-pred` is clever and correct but inconsistent with the rest of the app. Maybe if it wasn't so long and awkward lol.

Comment (merrickbrown) [reply src/clj/game/core/props.clj]: Done

**Pull request merged**